### ISSUE TITLE
Warp GUI text does not adhere to system preferences.

### ISSUE THREAD

**Issue by RokeJulianLockhart** [warpdotdev/Warp#4221]
### Discord username (optional)

https://linksta.cc/going/380957121?https://discord.com/users/798720532848705576

### Describe the bug

It's smaller than I've defined, and significantly more blurry.

> ![20240222T164335GMT](https://github.com/warpdotdev/Warp/assets/42837531/fb47588a-ad50-4fc7-8da9-f3f83d53791f)

### To reproduce

1.	Install `cpe:/o:opensuse:tumbleweed:20240220`.
1.	Install https://docs.warp.dev/getting-started/getting-started-with-warp#opensuse-and-sle-based-distributions using https://releases.warp.dev/stable/v0.2024.02.20.08.01.stable_01/warp-terminal-v0.2024.02.20.08.01.stable_01-1.x86_64.rpm.
1.	Invoke `/usr/share/applications/dev.warp.Warp.desktop`.

### Expected behavior

> ![20240222T164434GMT](https://github.com/warpdotdev/Warp/assets/42837531/f8bddd38-480a-4126-b3d7-02a6023acc93)

### Operating system

Linux

### Operating system and version

`cpe:/o:opensuse:tumbleweed:20240220`

### Shell Version

`sudo snap install powershell --channel=7.4.1/stable --classic`

### Current Warp version

`v0.2024.02.20.08.01.stable_01` from `warp-0.6.2-1.2.src`

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Recent working Warp date

Never.

### Additional context

https://docs.kde.org/stable5/en/plasma-workspace/kcontrol/fonts/index.html may be of use.

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Sorry about this @RokeJulianLockhart , i believe warp doesn't adhere to system fonts, so you'll have to see them in [Settings > Appearance > Text](https://docs.warp.dev/appearance/text-fonts-cursor). We recommend [nerd fonts](https://www.nerdfonts.com/). Also I noticed you have powershell, currently, Warp only supports [bash, zsh, or fish](https://docs.warp.dev/getting-started/using-warp-with-shells). 

Finally, we'll be tracking this in the feature request you opened here https://github.com/warpdotdev/Warp/issues/4227

Closing as warp doesn't support system-wide font settings yet.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/warpdotdev/Warp/issues/4221#issuecomment-1960070338

@dannyneira, that's an entirely different issue — the warp *GUI* (not terminal) *does* adhere to my *font*, but little else of my font configuration — it disregards the size and anti-aliasing, which are both significant accessibility concerns considering that all application GUIs should at least adhere to the system configuration *by default*, and certainly never override it without recourse in arbitrary manners.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/warpdotdev/Warp/issues/4221#issuecomment-1960134569

@dannyneira, the fact that you mention font aliases again causes me to wonder whether you understand what I refer to. I'm referring to the GUI of the Warp application, not the terminal. The terminal is entirely separate.

**Comment by dannyneira** [comment]
Right, I was just mentioning that since you had opened the other issue for it. My mistake if I got mixed up or didn't understand your issue correctly. but I see the issue is with the GUI of Warp not rendering the font correctly, I'm reopening. Try and tweak the setting for "Enforce minimum contrast" in `Settings > Appearance > Text` to see if that helps. 

![CleanShot 2024-02-22 at 14 43 01](https://github.com/warpdotdev/Warp/assets/16809145/e09770c0-4dd8-4dc6-866f-e1b4dfe032e5)

**Comment by RokeJulianLockhart** [comment]
> https://github.com/warpdotdev/Warp/issues/4221#event-11897080391

Thank you, @dannyneira.

> https://github.com/warpdotdev/Warp/issues/4221#issuecomment-1960368240

That preference doesn't appear to affect *anything*. Are any of the preferences under *Text* meant to affect anything except the terminal?

**Comment by dannyneira** [comment]
Darn, I was uncertain if that may help, but figured it was worth a shot so thanks for trying. 

Unfortunately, there is no way to change the GUI fonts/size/etc. We're tracking a request for that functionality here: https://github.com/warpdotdev/Warp/issues/1443

**Comment by dannyneira** [comment]
Duplicate of #1443

**Comment by RokeJulianLockhart** [comment]
> https://github.com/warpdotdev/Warp/issues/4221#issuecomment-1972018921

@dannyneira, I don't believe it is. Customization is entirely different — all software should adhere to the system-defined preferences by default. User customization would require an overhaul of the codebase, whereas this would merely necessitate that the current values not be hardcoded. I don't want to customize the GUI at all.

**Comment by dannyneira** [comment]
right, we'll post updates on this thread regarding your request for not adhering to system-wide settings.

**Comment by cbaxley** [comment]
> Darn, I was uncertain if that may help, but figured it was worth a shot so thanks for trying.
> 
> Unfortunately, there is no way to change the GUI fonts/size/etc. We're tracking a request for that functionality here: [#1443](https://github.com/warpdotdev/Warp/issues/1443)

They have been tracking this since 2023. They aren't going to fix it. With AI it would take a developer one day.

**Comment by cbaxley** [comment]
> > [#4221 (comment)](https://github.com/warpdotdev/Warp/issues/4221#issuecomment-3250378035)
> 
> [@cbaxley](https://github.com/cbaxley), that's just not true. Unit test coverage, having the PR reviewed, and ensuring that an LLM (if used – most developers don't rely upon such crutches for production code) take some time.
> 
> Not a few years’ worth, but it's evidently not of high priority. Regardless, even if it were true, it doesn't much help one, unless you've some genuinely actionable advice that the development team hasn't already heard.

So you are saying that you can change the gui font now? How?
